### PR TITLE
Add type hints for audioop module

### PR DESCRIPTION
**Pull request by jlaine**

### CODE REVIEW

**Comment by jlaine** [comment]
I'm not sure how to handle the CI failure: the documentation for `audioop` mentions that the "state" returned by `adpcm2lin`, `lin2adpcm`, `ratecv` are tuples, but it does not explicitly state the number of elements nor their type.

I *could* just look at what it contains, but it feels as though this "state" is meant to be handled as opaque. I would appreciate guidance as to how to proceed.

**Comment by JelleZijlstra** [comment]
I think the best option is to just peek at the implementation and specify what the tuple contains, then put it in a type alias. It's not uncommon that to write stubs you have to peek into the implementation.

Alternatively, you could just use `Tuple[..., Any]` or an opaque `NewType`.

**Comment by jlaine** [comment]
Thanks @JelleZijlstra I have defined `AdpcmState` and `RatecvState` type aliases as suggested, CI now passes.

File: stdlib/2and3/audioop.pyi [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def minmax(fragment: bytes, width: int) -> Tuple[int, int]: ...
+def mul(fragment: bytes, width: int, factor: float) -> bytes: ...
+def ratecv(
+    fragment: bytes,
+    width: int,
+    nchannels: int,
+    inrate: int,
+    outrate: int,
+    state: Optional[RatecvState],
+    weightA: Optional[int] = ...,
```
Comment (JelleZijlstra) [inline stdlib/2and3/audioop.pyi]: It doesn't look like these are really `Optional`; they default to integer values.

Comment (jlaine) [reply stdlib/2and3/audioop.pyi]: Thanks for catching this, it's now fixed.

**Comment by jlaine** [comment]
Thanks @JelleZijlstra !

**Pull request merged**